lib/opentok/client.rb

```diff
@@ -12,18 +12,20 @@ module OpenTok
   class Client
     include HTTParty
 
-    open_timeout 2 # Set HTTParty default timeout (open/read) to 2 seconds
-
     # TODO: expose a setting for http debugging for developers
     # debug_output $stdout
 
-    def initialize(api_key, api_secret, api_url, ua_addendum="")
+    attr_accessor :api_key, :api_secret, :api_url, :ua_addendum, :timeout_length
+
+    def initialize(api_key, api_secret, api_url, ua_addendum='', opts={})
       self.class.base_uri api_url
       self.class.headers({
-        "User-Agent" => "OpenTok-Ruby-SDK/#{VERSION}" + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}" + (ua_addendum ? " #{ua_addendum}" : "")        
+        "User-Agent" => "OpenTok-Ruby-SDK/#{VERSION}" + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}" + (ua_addendum ? " #{ua_addendum}" : "")
       })
       @api_key = api_key
       @api_secret = api_secret
+      @timeout_length = opts[:timeout_length] || 2
```

Comment:
With the name change to `timeout_length`, does this get respected by HTTParty? This is partially my lack of Ruby knowledge showing

Reply:
Can we also have the incoming option be named `timeout` for consistency with the other SDKs that will have this same change?

Reply:
The parameter `timeout_length` is the input for the value, the actual configuration for HTTParty happens with the setting of the `open_timeout`, which is a class method of HTTParty. The value in `timeout_length` is passed to that HTTParty method.

Since `timeout` is a [class method on Object](https://apidock.com/ruby/Object/timeout) (albeit deprecated), I'd rather refrain from using that reserved word as per our SDK specs.